packages/react-aria-components/docs/Calendar.mdx

```diff
@@ -699,6 +711,186 @@ function MyCalendarCell(props) {
+Calendar passes props to its child components, such as the heading and buttons, via their associated contexts. These contexts are exported so you can also consume them in your own custom components. This enables you to reuse existing components from your app or component library together with React Aria Components.
+
+<ContextTable components={['Heading', 'Button', 'Text']} docs={docs} />
+
+This example consumes from `HeadingContext` in an existing styled heading component to make it compatible with React Aria Components. The <TypeLink links={docs.links} type={docs.exports.useContextProps} /> hook merges the local props and ref with the ones provided via context by Calendar.
+
+```tsx
+import type {HeadingProps} from 'react-aria-components';
+import {HeadingContext, useContextProps} from 'react-aria-components';
+
+const MyCustomHeading = React.forwardRef((props: HeadingProps, ref: React.ForwardedRef<HTMLHeadingElement>) => {
+  // Merge the local props and ref with the ones provided via context.
+  ///- begin highlight -///
+  [props, ref] = useContextProps(props, ref, HeadingContext);
+  ///- end highlight -///
+
+  // ... your existing Heading component
+  return <h2 {...props} ref={ref} />;
+});
+```
+
+Now you can use `MyCustomHeading` within a `Calendar`, in place of the builtin React Aria Components `Heading`.
+
+```tsx
+<Calendar>
+  {/*- begin highlight -*/}
+  <MyCustomHeading />
+  {/*- end highlight -*/}
+  {/* ... */}
+</Calendar>
+```
+
 ### Hooks
 
-If you need to customize things even further, such as accessing internal state or customizing DOM structure, you can drop down to the lower level Hook-based API. See [useCalendar](useCalendar.html) for more details.
+If you need to customize things even further, such as accessing internal state or customizing DOM structure, you can drop down to the lower level Hook-based API. See [useCalendar](useCalendar.html) for more details. This example uses the `useCalendarGrid` hook to build a single week calendar view.
+
+```tsx example render=false export=true
+import type {CalendarGridProps} from 'react-aria-components';
+import type {CalendarState} from 'react-stately';
+import {useCalendarGrid} from 'react-aria';
+
+interface WeekCalendarGridProps extends CalendarGridProps {
+  state: CalendarState
+}
+
+function WeekCalendarGrid({state, ...props}: WeekCalendarGridProps) {
+  let {gridProps} = useCalendarGrid(props, state);
+
+  return (
+    <table {...gridProps}>
+      <tbody>
+        <tr>
+          {state.getDatesInWeek(0).map((date, i) => <CalendarCell key={i} date={date} />)}
+        </tr>
+      </tbody>
+    </table>
+  );
+}
+```
+
+`WeekCalendarGrid` can be used within a `Calendar` in place of the default `CalendarGrid` component.
+
+```tsx example
+<Calendar visibleDuration={{weeks: 1}} defaultValue={today(getLocalTimeZone())}>
+  {({state}) => (
+    <div className="week">
+      <Heading />
+      <Button slot="previous">◀</Button>
+      <WeekCalendarGrid state={state} />
```

Comment:
Might be nice to be able to access state via context inside WeekCalendarGrid, but that context is not exported at the moment. I would like to discuss a pattern for this, since many of our "internal" contexts are not exported and therefore it's hard to replace just one part of a composite component.

Reply:
We do some component states on the context already? Combobox? what is the pattern you were thinking which could define this?

I'm particularly wary of components with multiple states, such as RSP's TableView which has both table state and resize state. Those are two components in RAC, so I guess it'd be two different contexts?

Reply:
Would this be limited to state only or would it basically be exposing every internal context (aka: `InternalSelectContext, {state, valueProps, placeholder: props.placeholder}`)?

Reply:
Yeah I don't really want to expose all of the internal contexts. But we need something so it's possible to replace just one part of a composite component, like CalendarGrid in Calendar, or Row in Table, or Radio in RadioGroup, without having to rewrite the whole thing. Some of those are only states, which already have public interfaces in stately, but others have other stuff (like select).